apps/server/src/api-key/dto/api-key.dto.ts

```diff
@@ -0,0 +1,31 @@
+import { IsNotEmpty, IsString } from 'class-validator';
+import { ApiKeyPermission } from 'src/authentication/interfaces/api-key.interface';
+
+export class IdParamDto {
+  @IsString()
+  @IsNotEmpty()
+  id: string;
+}
+
+export class PostApiKeyRequestDto {
+  name: string;
+  permission: ApiKeyPermission;
+}
+
+export class PostApiKeyResponseDto {
+  id?: string;
+  token: string;
+}
+
+export class GetApiKeyRequestDto {
+  id?: string;
+  name: string;
+  created_at: Date;
+}
+
+export class CreateApiKeyDto {
+  token: string;
+  name: string;
+  permission: ApiKeyPermission;
+  userId: string;
+}
```

Comment:
Minor thing could you update this DTO for using `class-validator` decorators, and also adjust the swagger-docs  route for the GET method to ensure that each user can only access their own API keys.



```typescript
export enum ApiKeyPermission {
  FULL_ACCESS = API_KEY_PERMISSION_KEYS.FULL_ACCESS, // imported from  from 'API_KEY_PERMISSION_KEYS'
  SENDING_ACCESS = API_KEY_PERMISSION_KEYS.SENDING_ACCESS,
  READ_ACCESS = API_KEY_PERMISSION_KEYS.READ_ACCESS,
}
```


```typescript
// CreateApiKeyRequest Schema
export class PostApiKeyRequestDto {
  @IsString()
  @IsNotEmpty()
  name: string;

  @IsEnum(ApiKeyPermission)
  @IsNotEmpty()
  permission: ApiKeyPermission;

  @IsString()
  domainId: string;
}

// 200 Response for POST /apiKeys
export class PostApiKeyResponseDto {
  @IsString()
  @IsNotEmpty()
  id: string;

  @IsString()
  @IsNotEmpty()
  token: string;
}

// 200 Response for GET /apiKeys
export class GetApiKeysResponseDto {
  @IsString()
  id: string;

  @IsString()
  name: string;

  @IsDateString()
  created_at: string;
}

// Aggregate response wrapper for GET /apiKeys
export class GetApiKeysWrapperResponseDto {
  data: GetApiKeysResponseDto[]
}

// CreateApiKeyDto for internal processing, if needed
export class CreateApiKeyDto {
  @IsString()
  @IsNotEmpty()
  token: string;

  @IsString()
  @IsNotEmpty()
  name: string;

  @IsEnum(ApiKeyPermission)
  @IsNotEmpty()
  permission: ApiKeyPermission;

  @IsString()
  @IsNotEmpty()
  userId: string;
}
```